cmd/profiles.go

```diff
@@ -97,33 +98,51 @@ User profiles are not overwritten on upgrade of elastic-stack, and can be freely
 				return errors.Wrap(err, "error listing all profiles")
 			}
 
-			table := tablewriter.NewWriter(os.Stdout)
-			var profilestable = profileToList(profileList)
-
-			table.SetHeader([]string{"Name", "Date Created", "User", "Version", "Path"})
-			table.SetHeaderColor(
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-				twColor(tablewriter.Colors{tablewriter.Bold}),
-			)
-			table.SetColumnColor(
-				twColor(tablewriter.Colors{tablewriter.Bold, tablewriter.FgCyanColor}),
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-				tablewriter.Colors{},
-			)
-
-			table.SetAutoMergeCells(false)
-			table.SetRowLine(true)
-			table.AppendBulk(profilestable)
-			table.Render()
+			format, err := cmd.Flags().GetString(cobraext.ProfileFormatFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ProfileFromFlagName)
+			}
 
-			return nil
+			switch format {
+			case "table":
+				table := tablewriter.NewWriter(os.Stdout)
+				var profilestable = profileToList(profileList)
+
+				table.SetHeader([]string{"Name", "Date Created", "User", "Version", "Path"})
+				table.SetHeaderColor(
+					twColor(tablewriter.Colors{tablewriter.Bold}),
+					twColor(tablewriter.Colors{tablewriter.Bold}),
+					twColor(tablewriter.Colors{tablewriter.Bold}),
+					twColor(tablewriter.Colors{tablewriter.Bold}),
+					twColor(tablewriter.Colors{tablewriter.Bold}),
+				)
+				table.SetColumnColor(
+					twColor(tablewriter.Colors{tablewriter.Bold, tablewriter.FgCyanColor}),
+					tablewriter.Colors{},
+					tablewriter.Colors{},
+					tablewriter.Colors{},
+					tablewriter.Colors{},
+				)
+
+				table.SetAutoMergeCells(false)
+				table.SetRowLine(true)
+				table.AppendBulk(profilestable)
+				table.Render()
+
+				return nil
+			case "json":
+				data, err := json.Marshal(profileList)
+				if err != nil {
+					return errors.Wrap(err, "error listing all profiles in JSON format")
+				}
+				fmt.Print(string(data))
+				return nil
+			}
+
+			return errors.New("format " + format + " not supported")
```

Comment:
nit: you can move to `default:` of the switch above (just in case somebody accidentally defines a default behavior).